src/d3a/models/area/stats.py

```diff
@@ -131,20 +130,18 @@ def current_market(self):
         past_markets = list(self._markets.past_markets.values())
         return past_markets[-1] if len(past_markets) > 0 else None
 
-    def get_market_stats(self, market_slot_list, dso=False):
+    def get_last_market_stats(self, dso=False):
         out_dict = {}
-        for time_slot_str in market_slot_list:
-            try:
-                time_slot = from_format(time_slot_str, DATE_TIME_FORMAT, tz=TIME_ZONE)
-            except ValueError:
-                return {"ERROR": f"Time string '{time_slot_str}' is not following "
-                                 f"the format '{DATE_TIME_FORMAT}'"}
-            out_dict[time_slot_str] = copy(self.rate_stats_market.get(
-                time_slot, default_trade_stats_dict))
-            out_dict[time_slot_str]["market_bill"] = self._get_market_bills(time_slot)
-            if dso:
-                out_dict[time_slot_str]["area_throughput"] = \
-                    self._get_market_area_throughput(time_slot)
+        if self.current_market is None:
+            return out_dict
+        current_market_time_slot_str = self.current_market.time_slot_str
```

Comment:
I think you can skip this, the new name is not really shorter than the direct variable name :-)